raiden/network/proxies/token_network.py

```diff
@@ -1390,6 +1397,30 @@ def _set_total_withdraw(
                 if total_withdraw_done:
                     raise RaidenRecoverableError("Requested total withdraw was already performed")
 
+                if detail.state > ChannelState.OPENED:
+                    msg = (
+                        f"setTotalWithdraw failed because the channel closed "
+                        f"before the transaction was mined. "
+                        f"current_state={detail.state}"
+                    )
+                    raise RaidenRecoverableError(msg)
+
+                if detail.state < ChannelState.OPENED:
+                    msg = (
+                        f"setTotalWithdraw failed because the channel never "
+                        f"existed. current_state={detail.state}"
+                    )
+                    raise RaidenUnrecoverableError(msg)
+
+                if expiration_block < failed_at_blocknumber:
```

Comment:
It would be nice to have consistent ranges. The `settle_window` includes the last block, the `min` and `max` values for `settle_timeout` are inclusive, the token deposit limit includes the latest value, and the channel deposit limit is inclusive. It looks like the withdraw expiration is the only exclusive range.

Reply:
The expiration of hashtimelock is also like the withdraw expiration.